### PR TITLE
Add SIGUSR1 handler and example addition to the HTTP ingester.

### PR DESCRIPTION
**Pull request by david-fritz-gravwell**

### CODE REVIEW

File: debug/debug.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	if err != nil {
+		return
+	}
+
+	cpubuf := &bytes.Buffer{}
+	if err := pprof.StartCPUProfile(cpubuf); err == nil {
+		time.Sleep(CPU_SLEEP)
+		pprof.StopCPUProfile()
+		cpu.Write(cpubuf.Bytes())
+		cpu.Close()
```
Comment (kris-watts-gravwell) [inline debug/debug.go]: move this to a defer after successful Create

File: debug/debug.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	mem, err := os.Create(memName)
+	if err != nil {
+		return
+	}
+
+	membuf := &bytes.Buffer{}
+	runtime.GC()
+	if err := pprof.WriteHeapProfile(membuf); err == nil {
+		mem.Write(membuf.Bytes())
+		mem.Close()
```
Comment (kris-watts-gravwell) [inline debug/debug.go]: move this to a defer after create

File: debug/debug.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		if n < size {
+			break
+		}
+		size *= 2
+		if size >= MAX_STACK_SIZE {
+			return
+		}
+	}
+	st.Write(buf[:n])
+	st.Close()
```
Comment (kris-watts-gravwell) [inline debug/debug.go]: move this to a defer

**Review by kris-watts-gravwell — APPROVED** [review]
LGTM

**Review by john-floren-gravwell — APPROVED** [review]

**Pull request merged**